Gender distribution

SaraGomesES opened the pull request:
Evaluates the gender distribution regarding diagnosis, procedures, encounter classification and encounter description.

SaraGomesES commented:
Gender distribution by encounterclass

| ENCOUNTERCLASS | total_patients | total_male | total_female | male_proportion | female_proportion |
|----------------|----------------|------------|--------------|-----------------|-------------------|
| ambulatory     | 37668          | 22085      | 15583        | 58.63           | 41.37             |
| emergency      | 3441           | 1712       | 1729         | 49.75           | 50.25             |
| home           | 268            | 132        | 136          | 49.25           | 50.75             |
| hospice        | 146            | 79         | 67           | 54.11           | 45.89             |
| inpatient      | 1120           | 639        | 481          | 57.05           | 42.95             |
| outpatient     | 14371          | 6939       | 7432         | 48.28           | 51.72             |
| snf            | 167            | 83         | 84           | 49.7            | 50.3              |
| urgentcare     | 7193           | 4638       | 2555         | 64.48           | 35.52             |
| virtual        | 115            | 83         | 32           | 72.17           | 27.83             |
| wellness       | 28632          | 15237      | 13395        | 53.22           | 46.78             |

SaraGomesES commented:
Gender distribution by encounterclass description

| ENCOUNTERCLASS |                     DESCRIPTION                    | total_patients | total_male | total_female | male_proportion | female_proportion |
|----------------|----------------------------------------------------|----------------|------------|--------------|-----------------|-------------------|
| ambulatory     | Allergic disorder follow-up assessment             | 26             | 12         | 14           | 46.15           | 53.85             |
| ambulatory     | Allergic disorder initial assessment               | 47             | 25         | 22           | 53.19           | 46.81             |
| ambulatory     | Asthma follow-up                                   | 355            | 162        | 193          | 45.63           | 54.37             |
| ambulatory     | Discussion about treatment (procedure)             | 9              | 0          | 9            | 0.0             | 100.0             |
| ambulatory     | Domiciliary or rest home patient evaluation...     | 85             | 52         | 33           | 61.18           | 38.82             |
| ambulatory     | Encounter for Problem                              | 12             | 0          | 12           | 0.0             | 100.0             |
| ambulatory     | Encounter for check up                             | 873            | 447        | 426          | 51.2            | 48.8              |
| ambulatory     | Encounter for check up (procedure)                 | 1              | 0          | 1            | 0.0             | 100.0             |
| ambulatory     | Encounter for problem                              | 1613           | 798        | 815          | 49.47           | 50.53             |
| ambulatory     | Encounter for problem (procedure)                  | 21529          | 15650      | 5879         | 72.69           | 27.31             |

SaraGomesES commented:
Gender distribution by procedures 

| DESCRIPTION                                               | total_patients | total_male | total_female | male_proportion | female_proportion |
|-----------------------------------------------------------|----------------|------------|--------------|-----------------|-------------------|
| 12 lead electrocardiogram (procedure)                     | 4              | 3          | 1            | 75.0            | 25.0              |
| AFP test - antenatal                                      | 276            | 0          | 276          | 0.0             | 100.0             |
| Absolute CD4 count procedure (procedure)                  | 22             | 9          | 13           | 40.91           | 59.09             |
| Administration of anesthesia AND/OR sedation (procedure)  | 4              | 3          | 1            | 75.0            | 25.0              |
| Administration of anesthesia for procedure (procedure)    | 39             | 23         | 16           | 58.97           | 41.03             |
| Administration of diphtheria, pertussis, and tetanus vaccine | 278            | 0          | 278          | 0.0             | 100.0             |
| Admission to burn unit                                    | 3              | 1          | 2            | 33.33           | 66.67             |
| Admission to intensive care unit (procedure)              | 18             | 12         | 6            | 66.67           | 33.33             |
| Admission to long stay hospital                           | 3              | 1          | 2            | 33.33           | 66.67             |
| Admission to neurosurgical department                     | 1              | 1          | 0            | 100.0           | 0.0               |

rafawendel commented on gd-by-encounterclass.sql:
```diff
@@ -0,0 +1,12 @@
+count(*) as total_patients,
+sum(case when gender = 'M' then 1 else 0 end) as total_male,
+sum(case when gender = 'F' then 1 else 0 end) as total_female,
+round((sum(case when gender = 'M' then 1 else 0 end) * 100 / count(*)), 2) as male_proportion,
+round((sum(case when gender = 'F' then 1 else 0 end) * 100 / count(*)), 2) as female_proportion
```
✨

rafawendel reviewed (commented):
Everything is looking pretty much alright. The only blocker is having the CSVs instead of an updated database on which I could query. Can you fix that?

SaraGomesES commented:
> Everything is looking pretty much alright. The only blocker is having the CSVs instead of an updated database on which I could query. Can you fix that?

it's done! I removed the csv files and added the updated database version. #4

SaraGomesES commented:
> Gender distribution by diagnosis
> 
> diagnosis	total_patients	total_male	total_female	male_proportion	female_proportion
> Acute bacterial sinusitis (disorder)	13	10	3	76.92	23.08
> Acute bronchitis (disorder)	589	313	276	53.14	46.86
> Acute deep venous thrombosis (disorder)	46	30	16	65.22	34.78
> Acute pulmonary embolism (disorder)	44	19	25	43.18	56.82
> Acute viral pharyngitis (disorder)	12	6	6	50.0	50.0
> Alzheimer's disease (disorder)	26	5	21	19.23	80.77
> Anemia (disorder)	10311	7360	2951	71.38	28.62
> Asthma	42	0	42	0.0	100.0
> Atopic dermatitis	28	16	12	57.14	42.86
> Atrial fibrillation (disorder)	669	627	42	93.72	6.28

Due to the update of the 'medications_per_patient' table (https://github.com/SaraGomesES/promptly_duckdb/pull/4#discussion_r1307654282), there were some changes in the result of this query:

|                diagnosis                | total_patients | total_male | total_female | male_proportion | female_proportion |
|-----------------------------------------|----------------|------------|--------------|-----------------|-------------------|
| Acute bacterial sinusitis (disorder)    | 13             | 10         | 3            | 76.92           | 23.08             |
| Acute bronchitis (disorder)             | 502            | 267        | 235          | 53.19           | 46.81             |
| Acute deep venous thrombosis (disorder) | 8              | 4          | 4            | 50.0            | 50.0              |
| Acute pulmonary embolism (disorder)     | 7              | 4          | 3            | 57.14           | 42.86             |
| Acute viral pharyngitis (disorder)      | 12             | 6          | 6            | 50.0            | 50.0              |
| Alzheimer's disease (disorder)          | 26             | 5          | 21           | 19.23           | 80.77             |
| Anemia (disorder)                       | 34             | 22         | 12           | 64.71           | 35.29             |
| Asthma                                  | 2              | 0          | 2            | 0.0             | 100.0             |
| Atopic dermatitis                       | 16             | 10         | 6            | 62.5            | 37.5              |
| Atrial fibrillation (disorder)          | 18             | 12         | 6            | 66.67           | 33.33             |
